docs/data types.md

```diff
@@ -0,0 +1,277 @@
+
+```jsx
+// With decimals:
+let some_number = 34.00;
+
+// Without decimals:
+let another_number = 34;
+```
+
+### BigInt
+
+JavaScript BigInt is a new datatype that can be used to store integer values that are too big to be represented by a normal JavaScript Number.
+
+Example:
+
+```jsx
+let x = BigInt("123456789012345678901234567890");
+```
+
+### Boolean
+
+Booleans can only have two values: true or false.
+
+Example:
+
+```jsx
+let x = true;
+let y = false;
+```
+
+Additionally, boolean data can be returned from expressions
+
+Example:
+
+```jsx
+let x = 5;
+let y = 5;
+let z = 6;
+
+console.log(x == y)       // Returns true
+console.log(x == z)       // Returns false
+```
+
+### Undefined
+
+In JavaScript, a variable without a value, has the value undefined. The type is also undefined or you can directly assign the value to undefined.
+
+Example:
+
+```jsx
+let car;    // Value is undefined, type is undefined
+car = undefined;    // Value is undefined, type is undefined
+```
+
+### Null
+
+In JavaScript, null is an assigned value that represents the intentional absence of any object value. It is one of JavaScript's primitive values.
+
+Example:
+
+```jsx
+let person = null;
+```
+
+### Symbol
+
+A symbol is a unique and immutable primitive value and may be used as the key of an Object property.
+
+Example:
+
+```jsx
+const sym = Symbol('foo');
+```
+
+## Composite Data Types
+
+### Arrays
+
+Think of arrays as a collection of your favorite books. 1. You have a list of titles: "Harry Potter," "Percy Jackson," and "The Hunger Games." In JavaScript, you can use an array to store these book titles together.
+
+Items in arrays are called ***“Elements”***
+
+JavaScript arrays are written with square brackets. Elements are separated by commas.
+
+```jsx
+const my_book_collection = ["Harry Potter", "Percy Jackson", "The Hunger Games"];
+
+const cars = ["Saab", "Volvo", "BMW"];
+```
+
+*Note: the element can be of any type, it does not necessarily have to be string as shown in the example.*
+
+Array indexes are zero-based, which means the first item is [0], second is [1], and so on.
+
+In the above example:
+
+```jsx
+cars[0] // Saab
+cars[1] // Volvo
+cars[2] // BMW
+```
+
+Further explanations are available in the ***Arrays*** chapter!
+
+### Objects
+
+Consider a recipe for a sandwich. It has different parts: "ingredients," "steps," and "prep time." In JavaScript, you can create an object to hold all these details in a structured way.
+
+JavaScript objects are written with curly braces `{}`
+
+Object properties are written as name: value pairs, separated by commas.
+
+Example:
+
+```jsx
+const sandwich = {
+		name: "Ham Cheese Sandwich",
+		ingredients: ["bread", "ham", "cheese"],
+		steps: [......],
+		preptime: ....
```

Comment:
These can be mistaken for spread operators? Maybe add a comment to clarify?

Reply:
wdym 'spread' ?

Reply:
> wdym 'spread' ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax